### ISSUE TITLE
Missing notification when NZB fetch fails 

### ISSUE THREAD

**Issue by tokred** [nzbget/nzbget#157]
Just noticed a possible bug with v16.4 when NZB fetch failed, no post processing was performed and therefore SickRage was not notified of the failure:

```
...
Sun Jan 17 14:45:06 2016    INFO    SickRage connected to drop of XXXXXX any moment now.
Sun Jan 17 14:45:06 2016    INFO    Queue XXXXXX @ foo.com
Sun Jan 17 14:45:06 2016    DETAIL  Downloading XXXXXX @ foo.com
Sun Jan 17 14:45:07 2016    DETAIL  URL http://foo.com/bar redirected to https://foo.com/bar
Sun Jan 17 14:45:08 2016    DETAIL  Downloading XXXXXX @ foo.com
Sun Jan 17 14:45:11 2016    ERROR   Could not read from TLS-Socket: Success.
Sun Jan 17 14:45:11 2016    WARNING URL XXXXXX @ foo.com failed: Unexpected end of file
Sun Jan 17 14:45:11 2016    DETAIL  Waiting 10 sec to retry
Sun Jan 17 14:45:23 2016    DETAIL  Downloading XXXXXX @ foo.com
Sun Jan 17 14:45:25 2016    ERROR   Could not read from TLS-Socket: Success.
Sun Jan 17 14:45:25 2016    WARNING URL XXXXXX @ foo.com failed: Unexpected end of file
Sun Jan 17 14:45:25 2016    DETAIL  Waiting 10 sec to retry
Sun Jan 17 14:45:37 2016    DETAIL  Downloading XXXXXX @ foo.com
Sun Jan 17 14:45:40 2016    ERROR   Could not read from TLS-Socket: Success.
Sun Jan 17 14:45:40 2016    WARNING URL XXXXXX @ foo.com failed: Unexpected end of file
Sun Jan 17 14:45:40 2016    ERROR   Download XXXXXX @ foo.com failed
(EOF)
```

I am aware of issue #120 and that it will be fixed in 17.0. However this is another issue, as post processing via nzbToSickBeard was missing which results in the situation, that SickRage does not get notified of the failed download.

Is this a configuration issue or a bug? Would be happy for a solution.

**Comment by hugbug** [comment]
If fetch fails the pp-scripts are not executed because there is nothing to post-process. This situation differs from failed download because a fetch failure doesn't mean the files are not available on news server.

In most cases fetch failures are caused by misconfiguration somewhere (sickrage etc.) and require user intervention to fix the issue.

Saying that there is still a possibility for scripts to process fetch failures. The script must handle queue-event URL_COMPLETED and to do so the script must be declared as queue-script (in addition to being pp-script). For short description see [queue scrips](https://github.com/nzbget/nzbget/wiki/Queue-scripts). I'd be happy to provide more details via forum if needed.

As for the tls-error - what's your platform? Since you mentioned #120 I assume you compile yourself. Can you please test using nzbget installer provided on download page?

**Comment by tokred** [comment]
hugbug, thank you for your response. Your explanation makes sense to me, so a post processing script like nzbToSickBeard is not able to handle this particular error situation. Would nzbget handle a searcher-side outage (HTTP 404/500 or completely offline) in a similar way so the requesting SickRage would not able to initiate a download from another source?

I am most certainly ruling out misconfiguration as, apart from the above, my setup is working fine. My setup is based on FreeBSD where nzbget was installed via pkg (binary package manager). The package has been pre-compiled with GnuTLS (see [Freshports](http://www.freshports.org/news/nzbget/)), so I will switch to the port version and compile it with OpenSSL by myself.

**Comment by hugbug** [comment]
Do you have these errors all the time or random? If sickrage was able to fetch info via rss or api the indexer is available therefore. Strange that nzbget can't fetch the nzb-file. 

Testing nzbget with openssl is the first thing to do.

BTW, starting with v17 I will offer FreeBSD installer package (similar to the Linux one). If you want to test current development version send me a note to nzbget@gmail.com.

**Comment by tokred** [comment]
As already mentioned my nzbget setup works fine for all other downloads. This issue occured only once where two nzb files both from cdn.nzbgeek.info where enqueued by SickRage.

I recompiled nzbget with OpenSSL and this works flawless so far, also with nzb files from the aforementioned source.
